### PR TITLE
Check triggers cmake error unnecessarily

### PR DESCRIPTION
**Pull request by greg7mdp**
### Context of Change

When using cmake-gui to generate solution files on windows, the default platform is stated to be x64, so it is natural to leave the field unfilled, but even among the proposed choices, Win64 is nowhere to be found, we see only x64.

So whether we leave the field blank, or select x64 in the dropdown, cmake generation fails because it checks for the presence of `Win64`.

This change makes the generation of solution files more user friendly on windows, by not causing an error unless we find `Win32`. 

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

### CODE REVIEW

**Comment by greg7mdp** [comment]
image

**Comment by legleux** [comment]
For Visual Studio 2017 and prior, the default wasn't x64 so arch couldn't leave it blank.
I vote we don't even bother with 2017, but only use 2019 and 2022.

**Comment by greg7mdp** [comment]
>  I vote we don't even bother with 2017, but only use 2019 and 2022.

sounds good to me!

**Review by legleux — APPROVED** [review]

File: Builds/CMake/RippledSanity.cmake [hunk 1]
```diff
@@ -73,7 +73,7 @@ if ("${CMAKE_CURRENT_SOURCE_DIR}" STREQUAL "${CMAKE_BINARY_DIR}")
 endif ()
 
 if ("${CMAKE_GENERATOR}" MATCHES "Visual Studio" AND
-    NOT ("${CMAKE_GENERATOR}" MATCHES .*Win64.*))
+   ("${CMAKE_GENERATOR_PLATFORM}" MATCHES .*Win32.*))
   message (FATAL_ERROR
     "Visual Studio 32-bit build is not supported. Use -G\"${CMAKE_GENERATOR} Win64\"")
```
Comment (thejohnfreeman) [inline Builds/CMake/RippledSanity.cmake:78]: How about this instead?

```cmake
if (MSVC AND CMAKE_GENERATOR_PLATFORM STREQUAL "Win32")
  message (FATAL_ERROR
    "Visual Studio 32-bit build is not supported. Use -A x64")
```

Comment (greg7mdp) [reply Builds/CMake/RippledSanity.cmake:78]: Yes, looks better. I'll do a sanity check and update it.

Comment (greg7mdp) [reply Builds/CMake/RippledSanity.cmake]: @thejohnfreeman I did the change, removing the `-A x64` as it does not seem to work.

**Pull request closed without merging**